l10n_nl_tax_invoice_basis/models/account_move_line.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2017 Onestein (<https://www.onestein.eu>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+INVOICE_TYPES = [
+    'out_invoice',
+    'out_refund',
+    'out_receipt',
+    'in_invoice',
+    'in__refund',
+    'in_receipt',
+]
+
+
+class AccountMoveLine(models.Model):
+    _inherit = 'account.move.line'
+
+    l10n_nl_date_invoice = fields.Date(
+        compute='_compute_l10n_nl_date_invoice', store=True)
+
+    @api.depends(
+        'move_id', 'move_id.invoice_date', 'move_id.type', 'company_id',
+        'company_id.l10n_nl_tax_invoice_basis', 'company_id.country_id')
+    def _compute_l10n_nl_date_invoice(self):
+        for line in self:
+            company = line.move_id.company_id
+            is_invoice_basis = company.l10n_nl_tax_invoice_basis
+            is_nl = company.country_id == self.env.ref('base.nl')
+            is_invoice = line.move_id.type in INVOICE_TYPES
```

Comment:
Use `line.move_id.is_sale_document(include_receipts=True)` and `line.move_id.is_purchase_document(include_receipts=True)` instead.
Remove INVOICE_TYPES.